동아리 리스트 조회 API

rrgks6221 opened the pull request:
<!-- 지라 티켓 url -->
### JIRA
[QYOG-87](https://dongurami.atlassian.net/browse/QYOG-87)

<!-- 내용 (필수) -->
### Description

동아리 리스트 조회 api
관계되는 테이블도 함께 만들었음

<!-- 리뷰어가 리뷰하기전 알면 좋을 내용 (선택) -->
### To Reviewer

목데이터 시드도 함께 만듬

<!-- 참고한 레퍼런스 링크 (선택) -->
### Reference Link

<!-- 관련된 이슈 링크 (선택) -->
### Related Issue Link

<!-- 작업한 API (선택) -->
### API

| Method | Path      | 설명           | 작업(추가, 수정, 삭제) |
| ------ | --------- | -------------- | ---------------------- |
| GET    | /api/clubs | 동아리 리스트 조회 | 추가                   |


[QYOG-87]: https://dongurami.atlassian.net/browse/QYOG-87?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

rrgks6221 committed:
db/QYOG-87: club table

https://dongurami.atlassian.net/browse/QYOG-87

rrgks6221 committed:
db/QYOG-87: club history table

https://dongurami.atlassian.net/browse/QYOG-87

rrgks6221 committed:
db/QYOG-87: club tag table

https://dongurami.atlassian.net/browse/QYOG-87

rrgks6221 committed:
db/QYOG-87: club tag link table

https://dongurami.atlassian.net/browse/QYOG-87

rrgks6221 committed:
db/QYOG-87: club category table

https://dongurami.atlassian.net/browse/QYOG-87

rrgks6221 committed:
db/QYOG-87: club category link table

https://dongurami.atlassian.net/browse/QYOG-87

rrgks6221 committed:
db/QYOG-87: type

https://dongurami.atlassian.net/browse/QYOG-87

rrgks6221 committed:
feat/QYOG-87: 동아리 리스트 조회 api

https://dongurami.atlassian.net/browse/QYOG-87

rrgks6221 committed:
Merge branch 'develop' into feature/QYOG-87/QYOG-87:

https://dongurami.atlassian.net/browse/QYOG-87

hobiJeong commented on src/apis/clubs/constants/club.constant.ts:
```diff
@@ -0,0 +1,6 @@
+import { ClubDto } from '@src/apis/clubs/dto/club.dto';
+
+export const CLUB_ORDER_FIELD: readonly (keyof ClubDto)[] = [
```
club의 order field 같은 경우엔 id가 포함되지 않는 이유가 있을까요?

rrgks6221 replied:
실수입니다. 추가할게요

hobiJeong commented on migrations/club/1708774714811-club.ts:
```diff
@@ -0,0 +1,72 @@
+          {
+            name: 'status',
+            type: 'enum',
+            enum: ['pending', 'active', 'inactive'],
```
혹시 만들어둔 enum을 안쓰는 이유가 있을까요?

rrgks6221 replied:
마이그레이션에는 최대한 import를 안하는게 나을듯합니다.
절대경로에대한 지원도 안해주기도 하고요

hobiJeong commented on src/apis/club-tags/dto/club-tag.dto.ts:
```diff
@@ -0,0 +1,20 @@
+import { BaseDto } from '@src/dto/base.dto';
+import { ClubTag } from '@src/entities/ClubTag';
+
+export class ClubTagDto
```
실제 club_tag 테이블엔 updatedAt이 없는데 이렇게 되면 BaseDto를 extends 받아서 ClubTagDto에는 실제 테이블에는 존재하지 않는 updatedAt 필드가 들어가지 않을까요?

rrgks6221 replied:
그렇네요 Omit으로 제거할게요

hobiJeong commented on src/apis/clubs/dto/club.dto.ts:
```diff
@@ -0,0 +1,51 @@
+  status: ClubStatus;
+
+  @Exclude()
+  deletedAt: Date;
```
```suggestion
  deletedAt: Date | null;
```

hobiJeong commented on src/apis/clubs/dto/find-club-list-query.dto.ts:
```diff
@@ -0,0 +1,44 @@
+    description: '동아리 명',
+  })
+  @IsNotEmpty()
+  @IsString()
```
다 string 타입으로 들어 올텐데 IsString이 필요 없지 않을까요?

rrgks6221 replied:
네 제거하는게 나을듯합니다.

hobiJeong commented on src/apis/clubs/services/clubs.service.ts:
```diff
@@ -0,0 +1,99 @@
+    const clubIds = clubs.map((club) => club.id);
+
+    const [clubCategoryLinks, clubTagLinks] = await Promise.all([
+      this.clubCategoryLinkRepository.find({
+        where: {
+          clubId: In(clubIds),
+        },
+        relations: {
+          clubCategory: true,
+        },
+      }),
+      this.clubTagLinkRepository.find({
+        where: {
+          clubId: In(clubIds),
+        },
+        relations: {
+          clubTag: true,
```
depth가 2 이상이라서 이렇게 짜신건가요?

rrgks6221 replied:
주석에 적어놓은 category 및 tag 필터링때문에 그렇습니다.
하나의 쿼리로 조인해서 사용하게되면 tagId 같은거로 필터링할 때 동아리가 가지고 있는 모든 태그를 가져오지 못하고 필터링된 태그만 가져오기 때문입니다.

hobiJeong commented on src/entities/ClubHistory.ts:
```diff
@@ -0,0 +1,80 @@
+  name: string;
+
+  @Column('text', { name: 'introduce', comment: '동아리 소개' })
+  introduce: string;
```
해당 컬럼도 nullable 아닌가요?

rrgks6221 replied:
네 맞습니다. 수정할게요

hobiJeong commented on src/entities/ClubHistory.ts:
```diff
@@ -0,0 +1,80 @@
+    comment: '동아리 로고 경로',
+    length: 255,
+  })
+  logoPath: string;
```
해당 컬럼도 nullabe 아닌가요?

rrgks6221 replied:
네 맞습니다. 수정할게요

rrgks6221 commented:
@hobiJeong 
추가로 name 필드에 unique 제약조건 걸게요

rrgks6221 committed:
refactor/QYOG-87: 빠진 작업 추가

1. order field id 추가
2. club tag dto updatedAt 제거
3. club dto deletedAt nullable 추가
4. 동아리명 필터링 dto IsString 제거
5. entity nullable 추가

https://dongurami.atlassian.net/browse/QYOG-87

hobiJeong commented on src/apis/clubs/dto/find-club-list-query.dto.ts:
```diff
@@ -0,0 +1,43 @@
+import { CsvToOrder, Order } from '@src/dto/transformer/csv-to-order.decorator';
+import { IsPositiveInt } from '@src/dto/validator/is-positive-int.decorator';
+
+export class FindClubListQueryDto extends PageDto implements Partial<ClubDto> {
```
필터링중에 id 필드도 추가되는게 좋을 것 같습니다.

rrgks6221 replied:
리스트조회 필터링에는 id를 필수로 할 필요는 없을거같습니다. 차라리 ids 같은 필드를 받는거는 추후 추가될 수 있을거같고요.

hobiJeong commented on src/apis/clubs/dto/club.dto.ts:
```diff
@@ -0,0 +1,51 @@
+  @ApiProperty({
+    description: '동아리 소개',
+  })
+  introduce: string;
+
+  @ApiProperty({
+    description: '동아리 로고 path',
+  })
+  logoPath: string;
```
dto에서도 null 타입이 반영되는게 좋을 것 같습니다

rrgks6221 replied:
감사합니다.

rrgks6221 committed:
refactor/QYOG-87: club dto nullable field

https://dongurami.atlassian.net/browse/QYOG-87

hobiJeong reviewed (approved):
고생하셨습니다!
